File: frontend/src/types/GroundResource.ts
import type { ResourceStatusValue } from "../constants/ResourceStatus";

export interface GroundResource {
  id: number;
  resource_code: string;
  resource_type: string;
  location: string;
  availability_status: ResourceStatusValue;
  maintenance_due_at: string | null;
  owner_team: string;
  available_from: string | null;
  available_to: string | null;
}

File: backend/src/constructors/GroundResource.go
package constructors

import (
	"groundTurn/src/constants"
	"groundTurn/src/models"
)

// NewGroundResource builds a resource ledger row during seeding/admin create.
func NewGroundResource(code, resourceType, location, teamID string, status constants.ResourceStatus) *models.GroundResource {
	return &models.GroundResource{
		ResourceCode:       code,
		ResourceType:       resourceType,
		Location:           location,
		AvailabilityStatus: string(status),
		OwnerTeam:          teamID,
	}
}
